Utilities/bootstrap

```diff
@@ -555,15 +532,27 @@ def get_swiftpm_flags(args):
             "-Xlinker", "@executable_path/../../../../../SharedFrameworks",
         ])
 
-    # Add a rpath to find core swift libraries on linux and Android. We don't
-    # need a rpath for Darwin because the Swift libraries are present in the OS.
-    if platform.system() == 'Linux':
-        if 'ANDROID_DATA' in os.environ:
-            build_flags.extend(["-Xlinker", "-rpath=$ORIGIN/../lib/swift/android"])
-            # Don't use GNU strerror_r on Android.
-            build_flags.extend(["-Xswiftc", "-Xcc", "-Xswiftc", "-U_GNU_SOURCE"])
-        else:
-            build_flags.extend(["-Xlinker", "-rpath=$ORIGIN/../lib/swift/linux"])
+    # Add a relative rpath to find core Swift libraries on platforms where they
+    # aren't known to be present as part of the base OS.
+    target_info_json = call_output([args.swiftc_path, '-print-target-info'], verbose=args.verbose)
+    parsed_target_info = json.loads(target_info_json)
+    if not parsed_target_info["target"]["librariesRequireRPath"]:
+        platform_path = None
+        for path in parsed_target_info["paths"]["runtimeLibraryPaths"]:
+            platform_path = re.search(r'(lib/swift/[^/]+)$', path)
+            if platform_path:
+                if platform.system() == 'Darwin':
```

Comment:
We can just ignore Darwin here because it now has Swift in the OS and doesn't require run path if you're not deploying to older OSes.

Reply:
See my source links in the OP, this is only invoked for the older Darwin OS's where Swift _isn't_ in the OS.

Reply:
Ah. We can just remove it since SwiftPM doesn't deploy that far back on Darwin anymore.

Reply:
Done.